packages/block-editor/src/components/block-content-overlay/style.scss

```diff
@@ -18,23 +12,19 @@
 		z-index: z-index(".block-editor-block-content-overlay__overlay");
 	}
 
-	&:hover:not(.is-dragging-blocks),
-	&.parent-highlighted {
-		> .block-editor-block-content-overlay__overlay {
-			background: rgba(var(--wp-admin-theme-color--rgb), 0.1);
-			box-shadow: 0 0 0 $border-width var(--wp-admin-theme-color) inset;
-		}
+	&:hover:not(.is-dragging-blocks)::before,
+	&.parent-highlighted::before {
+		background: rgba(var(--wp-admin-theme-color--rgb), 0.1);
+		box-shadow: 0 0 0 $border-width var(--wp-admin-theme-color) inset;
 	}
 
-	&.overlay-active:not(.is-dragging-blocks) {
-		*:not(.block-editor-block-content-overlay__overlay) {
-			pointer-events: none;
-		}
+	&.overlay-active:not(.is-dragging-blocks)::before {
```

Comment:
This doesn't seem to be working as expected anymore, as limiting pointer events on the ::before doesn't block them from other elements as the extra element did.  Currently you can see the hover interactions with children blocks, and can select the children block bypassing the parent and overlay:

![block-interactions-under-overlay](https://user-images.githubusercontent.com/28742426/129071691-4682ecb8-b3de-4fd1-a190-970e9ce6ff3b.gif)

We could update this to make all children have no pointer events until the overlay is disabled:

```suggestion
	&.overlay-active:not(.is-dragging-blocks) * {
```

But that won't solve the issue with the popover inserter. 🤔   When the popover inserters are triggered, they both break the visual styles of the overlay and allow a way to bypass it altogether

![popover-inserter-issue](https://user-images.githubusercontent.com/28742426/129075474-f21ed749-a1e0-4717-a6a3-f795f2fc8c17.gif)